### ISSUE TITLE
Include filename/chunkID in the checksum calculation for snapshots

### ISSUE THREAD

**Issue by npepinpe** [camunda-cloud/zeebe#6666]
**Description**

Currently, checksum computation is done as follows:

- chunks are files, are read completely in memory, and the checksum is based on their contents. Two files with the same contents would be identified as the same, even if their file name is different.
- combined checksums for the whole snapshot is done by calculating on each individual checksum, not on the file's data. This can lead to inaccurate checksums when including such things as two files with the same contents, empty files (marker files, a special case of the same-contents), when the ordering is not guaranteed, etc.

We should:

1. Ensure the ordering in which we compute the snapshot checksum is guaranteed
2. Include file names in the checksum computation
3. Combined checksum should be done by essentially stitching the chunks together and then computing the whole checksum. This can be done by updating the CRC32 instance in small chunks without a big memory footprint.

I think this is a breaking change for 0.26 so we would not back port this, but should be done before 1.0.

**Tangent**

It also occurred to me that we may want to explore having SFV files instead, i.e. a manifest of each file and their checksum. This would avoid also computing the checksum of large files at runtime, which for large files can have some overhead. But this is out of scope, though something to consider - possibly we should think of a way we can migrate the way we do snapshots without breaking compatibility post 1.0.
